documentation/design-docs/dotnet-tools.md

```diff
@@ -396,6 +396,81 @@ CONVERT
       Writing:       ./trace.speedscope.json
       Conversion complete
 
+### dotnet-stack
+
+SYNOPSIS
+
+    dotnet-stack [options] [command] [<args>]
+
+OPTIONS
+
+    --version
+        Display the version of the dotnet-trace utility.
+
+    -h, --help
+        Show command line help
+
+COMMANDS
+
+    report         Displays stack traces for the target process
+
+REPORT
+
+    dotnet-stack report -p|--process-id <pid>
+                        -n|--name <process-name>
+                        [-h|--help]
+
+    Prints the managed stack from every thread in the target process
+
+    -h, --help
+        Show command line help
+
+    Examples:
+      > dotnet-stack report -p 1234
+      Thread (0x151c):
+          [Native Frames]
```

Comment:
Why is `[Native Frames]` shown at the top of evert stack trace?

Reply:
The app I generated that sample from was sleeping/waiting on all the threads, so all the threads are in native code.  That usually shows up as `UNMANAGED_CODE_TIME` in TraceEvent, but we decided in the spec PR to rephrase that as `[Native Frames]`.

Reply:
The spec is a bit inconsistent with respect to SOS:
- `Prints the managed stack from every thread in the target process`
- `[Native Frames]` are by definition not part of a managed stack
- SOS `clrstack` doesn't seem to print anything for these native frames.

For instance
```
> clrstack
OS Thread Id: 0x11d8 (0)
Child SP       IP Call Site
7E89C6AC 76f4d1a0 [HelperMethodFrame: 7e89c6ac] System.Threading.WaitHandle.WaitOneCore(IntPtr, Int32)
7E89C740 6EB00CCE System.Threading.WaitHandle.WaitOneNoCheck(Int32) [/_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.cs @ 149]
7E89C760 6EB00BB4 System.Threading.WaitHandle.WaitOne(Int32) [/_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitHandle.cs @ 119]
7E89C778 6F342ECE System.Diagnostics.ProcessWaitState.WaitForExit(Int32) [/_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs @ 412]
7E89C7D8 6F33BD56 System.Diagnostics.Process.WaitForExitCore(Int32) [/_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs @ 203]
7E89C7F0 6F33AE6E System.Diagnostics.Process.WaitForExit() [/_/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs @ 1365]
7E89C800 72D55B3C AppSharedCore.CoreDump.CoreDumpCreator.WriteDump()
7E89C900 72D4FB48 AppSharedCore.CoreDump.CoreDumpCreator.CreateDump()
7E89CA08 74117438 TestConsole.Program.IsPrime(Int32)
7E89CA48 74116DC2 TestConsole.Program.NthPrime(Int32)
7E89CA88 74116CBA TestConsole.Program.Main(System.String[])
```

Reply:
I guess even SOS is inconsistent.  `clrstack -i` does show `[NativeStackFrame]`